src/core/jit/model.rs

```diff
@@ -360,3 +379,54 @@ impl<Input> OperationPlan<Input> {
         self.flat.len()
     }
 }
+
+/// DirectiveAdapter helps to bridge the gap between async_graphql's
+/// ConstDirective and our custom Directive implementation.
+pub trait DirectiveAdapter {
+    fn name(&self) -> &str;
+    fn arguments(&self) -> impl Iterator<Item = (&str, &ConstValue)>;
+}
+
+impl DirectiveAdapter for ConstDirective {
+    fn name(&self) -> &str {
+        &self.name.node
+    }
+
+    fn arguments(&self) -> impl Iterator<Item = (&str, &ConstValue)> {
+        self.arguments
+            .iter()
+            .map(|(k, v)| (k.node.as_str(), &v.node))
+    }
+}
```

Comment:
We can create a `DirectiveAdapter::from(const_directive)` instead which performs a ZERO copy transformation.